modules/electrophysiology_browser/php/sessions.class.inc

```diff
@@ -47,51 +47,63 @@ class Sessions extends \NDB_Page
      */
     function _hasAccess(\User $user) : bool
     {
-        return ($user->hasPermission('electrophysiology_browser_view_allsites')
+        return (($user->hasPermission('electrophysiology_browser_view_allsites')
                 || ($user->hasCenter($this->timepoint->getCenterID())
-                && $user->hasPermission('electrophysiology_browser_view_site')
-            )
-        );
+                && $user->hasPermission('electrophysiology_browser_view_site'))
+            ) && $user->hasProject($this->candidate->getProjectID()));
```

Comment:
This doesn't look right.. why is the site checking the timepoint's site but the project is checking the candidate's project? Shouldn't they both be the timepoint's (site/project)?

Reply:
Yes, it should be checking for the timepoint's project. I assumed that the timepoint and candidate projects would match (since I'm getting the candidate related to that timepoint), but I will change the code to get the project ID directly from the timepoint variable instead.

Reply:
This was resolved in the latest commit. I tested on both Chrome and Firefox and it still correctly stops a user from seeing pages for projects it's not affiliated with.